### ISSUE TITLE
validateSearch run multiple times even in other routes

### ISSUE THREAD

**Issue by ruiaraujo012** [TanStack/router#1481]
### Describe the bug

I'm having a strange behavior with one route. The `validateSearch` is running multiple times even in other routes.
In the images I have my route tree, the logs of `validateSearch` on other route match, and the `validateSearch` logs in the route that should validate the search.
As part from that, even my `validateSearch` is setting the default params if not present, they aren't showing in the url, same #1184 
My route file:
```tsx
const IndividualRecordSearchSchema = v.object({
  tab: v.fallback(v.enum_(TAB_KEYS), DEFAULT_TAB_KEY),
  // eslint-disable-next-line sort-keys
  subTab: v.fallback(v.enum_(SUB_TAB_KEYS), DEFAULT_SUB_TAB_KEY),
});

type IndividualRecordSearch = v.Output<typeof IndividualRecordSearchSchema>;

export const Route = createFileRoute('/_area-reservada/area-reservada/minhas-credenciacoes/ficha-individual')({
  component: () => <IndividualRecordPage />,
  loader: ({ context }) => {
    context.breadcrumbs = [
      {
        label: 'home',
        path: '/area-reservada',
      },
      {
        label: 'myAccreditations',
        path: '/area-reservada/minhas-credenciacoes/credenciacoes',
      },
      {
        label: 'individualRecord',
        path: '/area-reservada/minhas-credenciacoes/ficha-individual',
      },
    ];
  },
  validateSearch: (search: Record<string, unknown>): IndividualRecordSearch => {
    const parsed = v.parse(IndividualRecordSearchSchema, search);

    console.log('parsed >> ', parsed);

    return parsed;
  },
});
```

### Your Example Website or App

https://stackblitz.com/edit/tanstack-router-r3gsvr?file=src%2Froutes%2F_layout%2F_layout-2%2Flayout-a.tsx

### Steps to Reproduce the Bug or Issue

Navigating between pages in the sandbox, you can see the logs in the console.

### Expected behavior

The `validateSearch` running only once on the route and never on other routes. As well as putting the search params in the url.

### Screenshots or Videos

![image](https://github.com/TanStack/router/assets/16304685/3c5d7c2d-f220-44cd-add2-e1dc89df61c8)
![image](https://github.com/TanStack/router/assets/16304685/d4b6c309-091c-4754-b4bb-000a3c2e9810)
![image](https://github.com/TanStack/router/assets/16304685/a351dcb1-e400-4acf-b641-3cad3450a209)

### Platform

- OS: macOS
- Browser: Arc
- Version: 1.28.4

**Comment by SeanCassiere** [comment]
I'm closing this in favour of tracking this in #1919 as it has a workable minimal reproduction to work off of.
